src/test/java/umk/mat/pajda/ProjektZespolowy/PatternPasswordValidatorTest.java

```diff
@@ -0,0 +1,49 @@
+package umk.mat.pajda.ProjektZespolowy;
+
+import jakarta.validation.ConstraintValidatorContext;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.test.context.ActiveProfiles;
+import umk.mat.pajda.ProjektZespolowy.validators.PatternPasswordValidator;
+
+@ActiveProfiles("tests")
+public class PatternPasswordValidatorTest {
+
+  private PatternPasswordValidator patternPasswordValidator;
+
+  @Mock private ConstraintValidatorContext mock;
+
+  @BeforeEach
+  public void setUp() {
+    MockitoAnnotations.openMocks(this);
+    patternPasswordValidator = new PatternPasswordValidator();
+  }
+
+  @Test
+  public void noUppercaseTest() {
+    Assertions.assertFalse(patternPasswordValidator.isValid("n1!", mock));
+  }
+
+  @Test
+  public void noLowercaseTest() {
+    Assertions.assertFalse(patternPasswordValidator.isValid("N1!", mock));
+  }
+
+  @Test
+  public void noDigitTest() {
+    Assertions.assertFalse(patternPasswordValidator.isValid("Nn!", mock));
+  }
+
+  @Test
+  public void noSpecialCharacterTest() {
```

Comment:
test names should indicate the purpose of a test, maybe sth like `shouldFailWhenNoSpecialCharactersPresent` or `failsWhenNoSpecialCharactersPresent`?

Reply:
ok thanks for info i will change it